### PR TITLE
Refactor Delta testInputDataSize

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Comment by electrum** [comment]
We still haven't answered the question of why we need this test. What value does it provide? Is that value worth all of this extra complexity?

This adds additional complication to the Hadoop removal project. I'd rather just remove the test.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java [hunk 1]
```diff
@@ -157,36 +158,41 @@ protected QueryRunner createQueryRunner()
...
-         * INSERT INTO test_timestamps VALUES
-         * (TIMESTAMP '0100-01-01 01:02:03', 1), (TIMESTAMP '1582-10-15 01:02:03', 2), (TIMESTAMP '1960-01-01 01:02:03', 3), (TIMESTAMP '2020-01-01 01:02:03', 4);
-         */
-        NON_TPCH_TABLES.forEach(table -> {
-            String resourcePath = "databricks/" + table;
-            registerTableFromResources(table, resourcePath, queryRunner);
-        });
-
-        return queryRunner;
+        try {
```
Comment (electrum) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:160]:
```java
try (queryRunner) {

**Review by electrum — DISMISSED** [review]
We don't want to combine the storage configuration across connectors, since with #17498 we need to have different configuration in tests between Hive and Delta Lake connectors. This is why I'd rather drop this test entirely.

**Comment by findepi** [comment]
> What value does it provide? Is that value worth all of this extra complexity?

verifies that Delta connector doesn't read more data then it should, using Hive connector as the baseline

> We don't want to combine the storage configuration across connectors

Hive and Delta need to be able to talk to S3. Hive and Delta need to be able to talk to Azure storage. Hive and Delta need to be able to talk to GCS.  So I expect configuration to be same, unless there is a compelling reason not to be the same.

The test is easy to adapt if eg configuration properties are different, but currently they are not, so I could prepare for that.
Once they are different, it's trivial to update the test accordingly.

**Comment by electrum** [comment]
The first commit looks good

**Review by electrum — CHANGES_REQUESTED** [review]
The properties are different in https://github.com/trinodb/trino/pull/17498. Eventually, they will be the same, but that won't happen until Hive is fully converted to the new file system interface. Having all of this infrastructure for a single insignificant test doesn't seem worthwhile.

Why is this test important? Under what type of scenario would the Delta Lake connector read more data? Could we directly test that it reads the expected amount of data? For example, by verifying that the bytes read are in known valid range.

We need to be judicious when writing tests. Our current approach of writing very high level tests for absolutely everything is expensive.

If this test didn't exist and someone proposed to add it using the current approach, would we accept it? Just because someone chose to write this test at some point in the past doesn't mean that we need to keep it forever.

**Comment by electrum** [comment]
> Once they are different, it's trivial to update the test accordingly.

It may be trivial to update the code, but the development process is absolutely non-trivial, since I can't run the tests locally. They have to run in the CI with secrets. So this one test will add hours to the development process for every round trip.

**Comment by findepi** [comment]
> Why is this test important? Under what type of scenario would the Delta Lake connector read more data?

In none, but we want to have test coverage for that.

> We need to be judicious when writing tests.

Agreed

Also, we need more testing coverage, especially for file systems x cloud storages.
We constantly struggle with problems around it, and experience shows we don't have enough test coverage. I think it's premature to remove tests just because there is some maintenance cost to them.


>  Our current approach of writing very high level tests for absolutely everything is expensive.

I wish we had tests for everything.

**Comment by electrum** [comment]
I think we can agree that this input size test is unrelated to the type of file system. Therefore, there’s no need to test it against all cloud storage. That’s complex and wasteful.

So, if you can refactor it as single local test that doesn’t depend on external resources, then we can keep it.

**Comment by findepi** [comment]
> I think we can agree that this input size test is unrelated to the type of file system.

i agree that ideally the implementation should behave the same for all file systems, and that we should be able to simplify the test.
but how do we verify that?

**Review by raunaqmorarka — APPROVED** [review]

**Comment by raunaqmorarka** [comment]
TBH this `testInputDataSize` is too basic to satisfy the intended aim of no regressions in data read compared to hive.
It's missing nested types, doesn't have projection pruning, doesn't have predicate pruning, uses `processedInputDataSize` rather than `physicalInputDataSize` (e.g. with projection pruning it's possible that the connector read everything but returned only relevant blocks to the engine, discarding the unnecessary columns).
It would be better to have a standalone test in delta to sanity test processedInputDataSize without relying on hive as a baseline.

**Comment by findepi** [comment]
> It would be better to have a standalone test in delta to sanity test processedInputDataSize without relying on hive as a baseline.

i don't think we have such baselines tests for hive either.
i think we always use some baseline (eg read with DF vs read with DF disabled, etc.)

**Comment by findepi** [comment]
I will merge this so that https://github.com/trinodb/trino/pull/17498 doesn't have to remove the test coverage. We can always improve later, replacing the test with something that we agree is better.

**Pull request merged**